src/extract_weather.py

```diff
@@ -0,0 +1,127 @@
+                   "surface_pressure", "pressure_msl"],
+        "models": "ukmo_seamless",
+        "current": ["temperature_2m", "relative_humidity_2m", "wind_speed_10m",
+                    "wind_gusts_10m", "precipitation", "cloud_cover"],
+        "timezone": "GMT",
+        "forecast_days": 7,
+    }
+    return client.weather_api(url, params=params)[0]
+
+
+def process_current_data(response):
+    """Prints current weather data from the API response"""
+    current = response.Current()
+    current_temperature_2m = current.Variables(0).Value()
+    current_relative_humidity_2m = current.Variables(1).Value()
+    current_wind_speed_10m = current.Variables(2).Value()
+    current_wind_gusts_10m = current.Variables(3).Value()
+    current_precipitation = current.Variables(4).Value()
+    current_cloud_cover = current.Variables(5).Value()
+
+    # Some of this could be used as metrics on the dashboard
+    print(
+        f"\nCurrent time: {datetime.fromtimestamp(current.Time(),
+                                                  UTC).strftime('%Y-%m-%d %H:%M:%S')}")
+    print(f"Current temperature_2m: {current_temperature_2m}")
+    print(f"Current relative_humidity_2m: {current_relative_humidity_2m}")
+    print(f"Current wind_speed_10m: {current_wind_speed_10m}")
+    print(f"Current wind_gusts_10m: {current_wind_gusts_10m}")
+    print(f"Current precipitation: {current_precipitation}")
+    print(f"Current cloud_cover: {current_cloud_cover}")
+
+
+def process_hourly_data(response) -> pd.DataFrame:
+    """Returns a dataframe containing 24 hours of weather information starting from 
+    the current hour, from an API response.
+    """
+    hourly = response.Hourly()
+    hourly_data = {"Date": pd.date_range(
+        start=pd.to_datetime(hourly.Time(), unit="s", utc=True),
+        end=pd.to_datetime(hourly.TimeEnd(), unit="s", utc=True),
+        freq=pd.Timedelta(seconds=hourly.Interval()),
+        inclusive="left"
+    )}
+    hourly_data["Temperature (°C)"] = hourly.Variables(0).ValuesAsNumpy()
+    hourly_data["Relative Humidity (%)"] = hourly.Variables(1).ValuesAsNumpy()
+    hourly_data["Precipitation (mm)"] = hourly.Variables(2).ValuesAsNumpy()
+    hourly_data["Wind Speed (km/h)"] = hourly.Variables(3).ValuesAsNumpy()
+    hourly_data["Wind Gusts (km/h)"] = hourly.Variables(4).ValuesAsNumpy()
+    hourly_data["Visibility (m)"] = hourly.Variables(5).ValuesAsNumpy()
+    hourly_data["Cloud Cover (%)"] = hourly.Variables(6).ValuesAsNumpy()
+
+    hourly_dataframe = pd.DataFrame(data=hourly_data)
+
+    # Determines current hour and slices the df from this point +24 hours
+    current_hour = datetime.now().hour
+    return hourly_dataframe.iloc[current_hour:current_hour+24]
+
+
+def convert_unix_timestamp(unix_timestamp: int) -> str:
+    """Returns unix timestamp as a datetime string"""
+    return datetime.fromtimestamp(
+        unix_timestamp, UTC).strftime('%Y-%m-%d %H:%M:%S')
```

Comment:
quibble: can this be on one line?
```suggestion
    return datetime.fromtimestamp(unix_timestamp, UTC).strftime('%Y-%m-%d %H:%M:%S')
```